neo/Network/P2P/Payloads/WitnessScope.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Neo.IO;
+using Neo.IO.Json;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class WitnessScope : ISerializable, ICloneable<WitnessScope>
+    {
+        public static readonly WitnessScope Global = new WitnessScope()
+        {
+            Type = WitnessScopeType.Global,
+            ScopeData = new byte[0]
+        };
+
+        public WitnessScopeType Type;
+        public byte[] ScopeData;
+
+        public int Size => 1 + ScopeData.GetVarSize();
+
+        void ISerializable.Deserialize(BinaryReader reader)
+        {
+            Type = (WitnessScopeType)reader.ReadByte();
+            ScopeData = reader.ReadVarBytes(65536);
+        }
+
+        void ISerializable.Serialize(BinaryWriter writer)
+        {
+            writer.Write((byte)Type);
+            writer.WriteVarBytes(ScopeData);
```

Comment:
@igormcoelho Why write when is not needed, and why is not a fixed size (uint160)?

Reply:
Not fixed because of group pubkey, isnt it 33 bytes? For serialization youre right, better leave it optional field instead of zero.